### ISSUE TITLE
pypi-dependencies.json not recognized by flatpak-builder 1.0.8

### ISSUE THREAD

**Issue by leinardi** [flatpak/flatpak-builder-tools#61]
I used `flatpak-pip-generator` to generate a `pypi-dependencies.json` file containing all the `pip` dependencies but when I try to add it to the manifest, like indicated in the readme, i get this error:

```
(flatpak-builder:30833): flatpak-builder-ERROR **: 16:24:40.089: Failed to load included manifest (/home/leinardi/Workspace/gitlab/gwe/build/flatpak/pypi-dependencies.json): Expecting a JSON object, but the root node is of type “JsonArray”
```

This is how my module section looks like:
```json
  "modules": [
    "shared-modules/libappindicator/libappindicator-gtk3-introspection-12.10.json",
    "libdazzle.json",
    "python3-injector.json",
    "python3-setuptools.json",
    "python3-python-dateutil.json",
    "pypi-dependencies.json",
    {
      "name": "gwe",
      "buildsystem": "meson",
      "config-opts": [
        "--buildtype=release"
      ],
      "sources": [
        {
          "type": "git",
          "url": "https://gitlab.com/leinardi/gwe.git",
          "tag": "0.12.3",
          "commit": "0d8ccc6a1e73d8e36ffca08e42c0754d93b9f1e6"
        }
      ]
    }
  ]
```
This is the content of the `pypi-dependencies.json`:
```json
[
    {
        "name": "python3-injector",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} injector==0.17.0"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/2a/d1/0dfdf9cbc160883d408d1f610ac75e0b10f0af5da74e4f9d5d9d4b2c38f7/injector-0.17.0.tar.gz",
                "sha256": "882e91f8b4d693cfe29aa4eead3055480d7e2a6fb80a6a2c3b795650db720702"
            }
        ]
    },
    {
        "name": "python3-matplotlib",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} matplotlib==3.1.1"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/12/d1/7b12cd79c791348cb0c78ce6e7d16bd72992f13c9f1e8e43d2725a6d8adf/matplotlib-3.1.1.tar.gz",
                "sha256": "1febd22afe1489b13c6749ea059d392c03261b2950d1d45c17e3aed812080c93"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/c2/4b/137dea450d6e1e3d474e1d873cd1d4f7d3beed7e0dc973b06e8e10d32488/cycler-0.10.0.tar.gz",
                "sha256": "cd7b2d1018258d7247a71425e9f26463dfb444d411c39569972f4ce586b0c9d8"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/16/e7/df58eb8868d183223692d2a62529a594f6414964a3ae93548467b146a24d/kiwisolver-1.1.0.tar.gz",
                "sha256": "53eaed412477c836e1b9522c19858a8557d6e595077830146182225613b11a75"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/dd/bf/4138e7bfb757de47d1f4b6994648ec67a51efe58fa907c1e11e350cddfca/six-1.12.0.tar.gz",
                "sha256": "d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/ad/99/5b2e99737edeb28c71bcbec5b5dda19d0d9ef3ca3e92e3e925e7c0bb364c/python-dateutil-2.8.0.tar.gz",
                "sha256": "c89805f6f4d64db21ed966fda138f8a5ed7a4fdbc1a8ee329ce1b74e3c74da9e"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/d3/4b/f9f4b96c0b1ba43d28a5bdc4b64f0b9d3fbcf31313a51bc766942866a7c7/numpy-1.16.4.zip",
                "sha256": "7242be12a58fec245ee9734e625964b97cf7e3f2f7d016603f9e56660ce479c7"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/5d/3a/24d275393f493004aeb15a1beae2b4a3043526e8b692b65b4a9341450ebe/pyparsing-2.4.0.tar.gz",
                "sha256": "1873c03321fc118f4e9746baf201ff990ceb915f433f23b395f5580d1840cb2a"
            }
        ]
    },
    {
        "name": "python3-peewee",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} peewee==3.9.6"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/5e/66/5902f91abec78be8ef41eff4e547fe595c9e7b6c20f4be3d8cfb8e079061/peewee-3.9.6.tar.gz",
                "sha256": "bc879e1ffb7f7684f90d9dd236c77b5042118d079f94764f35e3806948829ac8"
            }
        ]
    },
    {
        "name": "python3-py3nvml",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} py3nvml==0.2.3"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/ee/63/59bf7e15b81c038b30931057cec2f3ce1d038a43be7b5b36809daa77229b/py3nvml-0.2.3.tar.gz",
                "sha256": "0a7b32e01c4d94142e9d44d826cd3b9637f86aa4d4e0c18e26464aa1cb9949bf"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/58/40/0d783e14112e064127063fbf5d1fe1351723e5dfe9d6daad346a305f6c49/xmltodict-0.12.0.tar.gz",
                "sha256": "50d8c638ed7ecb88d90561beedbf720c9b4e851a9fa6c47ebd64e99d166d8a21"
            }
        ]
    },
    {
        "name": "python3-pygobject",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} pygobject"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/40/ee/8daf797cda4ffe7724b9d47cbb5e2b44ca6235764622d163c586bd9bd7c5/PyGObject-3.32.2.tar.gz",
                "sha256": "fa82746df059b9e0dcf473e0a9ff0693aa5e064f1ca1e2c46b5ac815bceda1ad"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/48/20/5e83af98eb897935bf7dc39455e892ba866feebb9b7c3b392982866f9958/pycairo-1.18.1.tar.gz",
                "sha256": "70172e58b6bad7572a3518c26729b074acdde15e6fee6cbab6d3528ad552b786"
            }
        ]
    },
    {
        "name": "python3-pyxdg",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} pyxdg==0.26"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/47/6e/311d5f22e2b76381719b5d0c6e9dc39cd33999adae67db71d7279a6d70f4/pyxdg-0.26.tar.gz",
                "sha256": "fe2928d3f532ed32b39c32a482b54136fe766d19936afc96c8f00645f9da1a06"
            }
        ]
    },
    {
        "name": "python3-requests",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} requests"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/ad/13/eb56951b6f7950cadb579ca166e448ba77f9d24efc03edd7e55fa57d04b7/idna-2.8.tar.gz",
                "sha256": "c357b3f628cf53ae2c4c05627ecc484553142ca23264e593d327bcde5e9c3407"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/01/62/ddcf76d1d19885e8579acb1b1df26a852b03472c0e46d2b959a714c90608/requests-2.22.0.tar.gz",
                "sha256": "11e007a8a2aa0323f5a921e9e6a2d7e4e67d9877e85773fba9ba6419025cbeb4"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/fc/bb/a5768c230f9ddb03acc9ef3f0d4a3cf93462473795d18e9535498c8f929d/chardet-3.0.4.tar.gz",
                "sha256": "84ab92ed1c4d4f16916e05906b6b75a6c0fb5db821cc65e70cbd64a3e2a5eaae"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/c5/67/5d0548226bcc34468e23a0333978f0e23d28d0b3f0c71a151aef9c3f7680/certifi-2019.6.16.tar.gz",
                "sha256": "945e3ba63a0b9f577b1395204e13c3a231f9bc0223888be653286534e5873695"
            },
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/4c/13/2386233f7ee40aa8444b47f7463338f3cbdf00c316627558784e3f542f07/urllib3-1.25.3.tar.gz",
                "sha256": "dbe59173209418ae49d485b87d1681aefa36252ee85884c31346debd19463232"
            }
        ]
    },
    {
        "name": "python3-Rx",
        "buildsystem": "simple",
        "build-commands": [
            "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} Rx==3.0.1"
        ],
        "sources": [
            {
                "type": "file",
                "url": "https://files.pythonhosted.org/packages/9b/27/87ab7fc16a00141e0570ef8fc38c84f09e0840c9ce38ea7f1a79a9b32bcb/Rx-3.0.1.tar.gz",
                "sha256": "c7d168618e3cec35fda9c9c8b5d7f966739df4b1a99b315eae1fc2d7d69e2512"
            }
        ]
    }
]
```

**Comment by TingPing** [comment]
The contents of pypi-dependencies should not be in a list. Short term just remove the `[` and `]`. If thats how it came out its a bug (probably recent regression?).

**Comment by leinardi** [comment]
Yeah, I just noticed that it generate a list even if you use it for a single pip dependency, so without using the `--requirements-file` like I did in the OP.

**Comment by TingPing** [comment]
CC @bilelmoussaoui

**Comment by bilelmoussaoui** [comment]
I can confirm it's a regression I've caused in my latest PR. I have fixed that with #62.
